registry/remote/repository.go

```diff
@@ -614,6 +615,9 @@ type blobStore struct {
 
 // Fetch fetches the content identified by the descriptor.
 func (s *blobStore) Fetch(ctx context.Context, target ocispec.Descriptor) (rc io.ReadCloser, err error) {
+	if isForeignLayer(target) {
+		return nil, fmt.Errorf("the artifact with foreign layer %s is %w", target.Digest, errdef.ErrUnsupported)
```

Comment:
Actually, I think "artifact with foreign layer" looks a bit strange for the `blobStore.Fetch` method... 🤔
What if
```suggestion
		return nil, fmt.Errorf("%s: %s: foreign layer is unsupported: %w", target.Digest, target.MediaType, errdef.ErrUnsupported)
```
？

Reply:
The default media type of a blob cannot be foreign, there MUST be an artifact whose manifest is pointing to the to-be-fetched layer.